azure/services/managedclusters/managedclusters.go

```diff
@@ -178,8 +150,7 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		}
 	}
 
-	for i := range managedClusterSpec.AgentPools {
-		pool := managedClusterSpec.AgentPools[i]
+	for _, pool := range managedClusterSpec.AgentPools {
```

Comment:
This will result into an error if we pass in multiple system node pools. The name of each profile is a pointer to the pool.Name. This will eventually lead to creating profiles with mutiple pools with same name,sku,osdisksize etc... The create api will return error eventually.

Reply:
This was probably changed when I rebased my branch. Reverted the change now.